src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { type FC } from 'react';
+
+import searchIcon from './assets/search-icon.svg';
+import { DEFAULT_PLACEHOLDER } from './SearchBar.constants';
+import { useSearchBar } from './SearchBar.hook';
+import { SearchContainer, StyledInput, SearchIconButton } from './SearchBar.styles';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const SearchBar: FC<SearchBarProps> = (props) => {
+  const {
+    placeholder = DEFAULT_PLACEHOLDER,
+    inputOnChange,
+    onSearch,
+    debounceDelay,
+    value,
+    defaultValue,
+    onChange,
+    disabled,
+    ...otherProps
+  } = props;
+
+  const hookProps = {
+    onSearch,
+    debounceDelay,
+    value,
+    defaultValue,
+    onChange,
+  };
+
+  const {
+    value: inputValue,
+    handleInputChange,
+    handleKeyDown,
+    handleButtonClick,
+  } = useSearchBar(hookProps);
+
+  const combinedHandleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    // Type assertion para HTMLInputElement
+    const inputEvent = e as React.ChangeEvent<HTMLInputElement>;
+    handleInputChange(inputEvent);
+
+    if (inputOnChange) {
+      inputOnChange(e);
+    }
+  };
+
+  return (
+    <SearchContainer data-testid="search-bar-container">
+      <SearchIconButton
+        onClick={handleButtonClick}
+        aria-label="search"
+        data-testid="search-button"
+        disabled={disabled}
+      >
+        <img src={searchIcon} alt="Search" data-testid="search-icon" />
+      </SearchIconButton>
+      <StyledInput
+        value={inputValue}
+        onChange={combinedHandleChange}
+        placeholder={placeholder}
+        variant="standard"
+        inputProps={{
+          'aria-label': 'Search content',
```

Comment:
The input's aria-label and other strings are hardcoded in the component. Project i18n rules require strings to be localizable / not hardcoded. Provide these strings via props or a translation function.

**Sugerencia:**
```typescript
Accept an `ariaLabel` prop and/or use an i18n function: inputProps={{ 'aria-label': ariaLabel ?? t('searchContent') }}. Also localize placeholder if needed.
```